torch/csrc/jit/tensorexpr/types.h

```diff
@@ -31,6 +31,42 @@ enum class ScalarType : int8_t {
   NumOptions
 };
 
+static inline const char* toString(ScalarType t) {
+#define DEFINE_CASE(_, name) \
+  case ScalarType::name:     \
+    return #name;
+
+  switch (t) {
+    AT_FORALL_SCALAR_TYPES_WITH_COMPLEX_AND_QINTS(DEFINE_CASE)
+    DEFINE_CASE(_, Undefined)
+    DEFINE_CASE(_, Handle)
+    DEFINE_CASE(_, Uninitialized)
+    DEFINE_CASE(_, None)
+    DEFINE_CASE(_, NumOptions)
```

Comment:
NumOptions is not really a type, it's just the bottom of the enum (ie. the number of items in it). I'm not sure it makes sense to serialize it.